src/goto-harness/function_call_harness_generator.cpp

```diff
@@ -42,6 +43,12 @@ struct function_call_harness_generatort::implt
   recursive_initialization_configt recursive_initialization_config;
   std::unique_ptr<recursive_initializationt> recursive_initialization;
 
+  std::set<irep_idt> function_parameters_to_treat_as_arrays;
+  std::set<irep_idt> function_arguments_to_treat_as_arrays;
```

Comment:
I don't particularly enjoy the solution we have here - basically, we need to keep track of the names of function _arguments_ as declared on the function, and their mapping to _parameters_, and which _parameters_ the array initialisation applies to etc...

Reply:
This is just a note, we don't need to do it right now.